### PR TITLE
build specific service, recreate only modified service...

### PR DESCRIPTION
**Pull request by flobz**
This PR contains several improvements:
- build specific service
- up specific service
- compose-up recreate only modified service
- add compose_exec command
- transfer_service_status exit if podman cmd return 1

### CODE REVIEW

File: podman_compose.py [hunk 1]
```diff
@@ -1038,19 +1047,47 @@ def create_pods(compose, args):
...
+    if create:
+        compose.podman.run(podman_args)
+
 
 @cmd_run(podman_compose, 'up', 'Create and start the entire stack or some of its services')
 def compose_up(compose, args):
-    if args.services:
-        return up_specific(compose, args)
+    #if args.services:
+     #   return up_specific(compose, args)
```
Comment (SuperSandro2000) [inline podman_compose.py:1090]: Can this be removed?

File: podman_compose.py [hunk 2]
```diff
@@ -1302,13 +1369,26 @@ def compose_ps_parse(parser):
     parser.add_argument("-q", "--quiet",
         help="Only display container IDs", action='store_true')
 
+@cmd_parse(podman_compose, 'exec')
+def compose_exec_parse(parser):
+    parser.add_argument('services', metavar='services', nargs=1,
+                        help='services to push')
+    parser.add_argument('cmd', metavar='command', nargs=argparse.REMAINDER,
+                        help='command to execute')
+    parser.add_argument("-u","--user",metavar='USER',
```
Comment (SuperSandro2000) [inline podman_compose.py:1378]:
```suggestion
    parser.add_argument("-u", "--user", metavar='USER',
```

File: podman_compose.py (hunk 1 above)
Comment (flobz) [reply podman_compose.py]: yes, I commit changes

**Comment by flobz** [comment]
> Why do I need to comment on those GitHub?

**Comment by jerdna-regeiz** [comment]
Partly duplicate of https://github.com/containers/podman-compose/pull/132.
Having multiple features in one pull request makes it harder to review and integrate.
If this is going forward, could you copy the tests from https://github.com/containers/podman-compose/pull/132?

**Comment by flobz** [comment]
> Partly duplicate of #132.
> Having multiple features in one pull request makes it harder to review and integrate.


I wil make multiple pull request next time.
> If this is going forward, could you copy the tests from #132?

I revert the commit to avoid the duplication with #132 .

File: podman_compose.py [hunk 3]
```diff
@@ -1380,8 +1378,7 @@ def compose_build_parse(parser):
         help="attempt to pull a newer version of the image", action='store_true')
     parser.add_argument("--pull-always",
         help="attempt to pull a newer version of the image, Raise an error even if the image is present locally.", action='store_true')
-    parser.add_argument("--build-arg",metavar="KEY=VAL",
-                            help="Set build-time variables for services.", action='append')
+
```
Comment (SuperSandro2000) [inline podman_compose.py:1381]: I think it would be really nice if you would just drop this and the adding commit.

Comment (flobz) [reply podman_compose.py:1381]: I don't understand. Can you explain please ?

Comment (SuperSandro2000) [reply podman_compose.py:1381]: You rebase with git and just drop the original commit which added those lines in https://github.com/containers/podman-compose/pull/142/commits/e5c04a607cb5c5f83073fe5ff9a1dddfaba8da2e and the one removing it.

Comment (flobz) [reply podman_compose.py:1381]: ok thanks, I did it.

Comment (SuperSandro2000) [reply podman_compose.py:1381]: I have no idea what you did but you somehow amended the commits all again instead of overwriting them.

Comment (flobz) [reply podman_compose.py:1381]: Yes something goes wrong. I fix it. The build arg commit is deleted

**Comment by muayyad-alsadi** [comment]
I've merged the build args PR which might made this conflict.

I'll close this until conflicts are resolved.
feel free to open new fresh PRs.
one PR per issue.

**Comment by muayyad-alsadi** [comment]
BTW the below are very important.

-  up specific service
- compose-up recreate only modified service

thank you.

**Pull request closed without merging**